app/scripts/views/mixins/signin-mixin.js

```diff
@@ -105,16 +105,10 @@ define(function (require, exports, module) {
 
         if (verificationReason === VerificationReasons.SIGN_IN &&
             verificationMethod === VerificationMethods.EMAIL) {
-          return this.navigate('confirm_signin', {
-            account: account,
-            flow: this.flow
-          });
-        } else {
-          return this.navigate('confirm', {
-            account: account,
-            flow: this.flow
-          });
+          return this.navigate('confirm_signin', { account });
```

Comment:
Yep, it was being propagated between views so that flow events would work everywhere we needed them to. That act of explicitly passing it between views is what led to this refactor. The changes were getting out of hand and, not sure if you remember, I tried to refactor it to the base view but couldn't get that to work. Then, *BANG!*, flash of inspiration, I realised we could lean on events to make the pain go away.